### DISCUSSION TITLE
Trying to setup CoKreat program service locally.

### DISCUSSION THREAD

**Discussion by shubhambansaltarento** [Sunbird-coKreat/Community#70] [Q&A] [answered]
Hi @vaivk369, I am trying to setup program services locally but could not find any official end to end documentation though I have installed postgres DB and done the following configuration's (screen shot attached ) but still I am not able to create program locally, Kindly help in resolving the issue.
![Screenshot from 2023-04-04 14-37-53](https://user-images.githubusercontent.com/127090959/229747663-ac3a3fc1-20e3-4569-a844-ad578db39df4.png)

**Comment by vinukumar-vs** [comment]
@shubhambansaltarento Can you share the console logs if any thing is printing?

@vaivk369 @arun1205 @manimegalai-smohan

**Reply by shubhambansaltarento** [reply]
I am adding the curl request and the console screen shot 
CURL Request 

curl --location 'http://localhost:6000/program/v1/create' \
--header 'Content-Type: application/json' \
--header '' \
--header 'Accept: application/json' \
--header 'Accept-Language: en-GB,en;q=0.9' \
--data '{
    "request": {
        "program_id": "70b713d0-cc5b-11ed-a249-d1c2d4936c48",
        "name": "cokreat1Q",
        "description": "cokreat1Q",
        "type": "public",
        "collection_ids": null,
        "content_types": [],
        "target_collection_category": null,
        "targetprimarycategories": [],
        "targetprimarycategorynames": null,
        "target_type": "collections",
        "startdate": "2023-03-27T04:54:23.161Z",
        "enddate": "2023-06-22T18:29:59.000Z",
        "nomination_enddate": "2023-04-05T18:29:59.000Z",
        "shortlisting_enddate": "2023-05-06T18:29:59.000Z",
        "content_submission_enddate": "2023-06-04T18:29:59.000Z",
        "image": null,
        "status": "Draft",
        "slug": "sunbird",
        "config": {
            "board": [],
            "roles": [
                {
                    "id": 1,
                    "name": "CONTRIBUTOR",
                    "tabs": [
                        1
                    ],
                    "default": true,
                    "defaultTab": 1
                },
                {
                    "id": 2,
                    "name": "REVIEWER",
                    "tabs": [
                        2
                    ],
                    "defaultTab": 2
                }
            ],
            "header": {
                "id": "ng.sunbird.header",
                "ver": "1.0",
                "data": {},
                "author": "Venkat",
                "compId": "collectionComponent",
                "config": {
                    "tabs": [
                        {
                            "index": 1,
                            "label": "Contribute",
                            "onClick": "collectionComponent"
                        },
                        {
                            "index": 2,
                            "label": "Review",
                            "onClick": "collectionComponent"
                        },
                        {
                            "index": 3,
                            "label": "Dashboard",
                            "onClick": "dashboardComponent"
                        }
                    ]
                },
                "description": "",
                "publishedDate": ""
            },
            "medium": [],
            "subject": [],
            "_comments": "",
            "framework": [],
            "components": [
                {
                    "id": "ng.sunbird.collection",
                    "ver": "1.0",
                    "data": {},
                    "author": "Venkat",
                    "compId": "collectionComponent",
                    "config": {
                        "status": [
                            "Draft",
                            "Live"
                        ],
                        "filters": {
                            "explicit": [
                                {
                                    "code": "gradeLevel",
                                    "label": "Class",
                                    "range": [
                                        "Kindergarten",
                                        "Grade 1",
                                        "Grade 2",
                                        "Grade 3"
                                    ],
                                    "visibility": true,
                                    "multiselect": false,
                                    "defaultValue": [
                                        "Kindergarten",
                                        "Grade 1"
                                    ]
                                },
                                {
                                    "code": "subject",
                                    "label": "Subject",
                                    "range": [
                                        "English",
                                        "Mathematics",
                                        "Hindi"
                                    ],
                                    "visibility": true,
                                    "multiselect": false,
                                    "defaultValue": [
                                        "English"
                                    ]
                                }
                            ],
                            "implicit": [
                                {
                                    "code": "framework",
                                    "label": "Framework",
                                    "defaultValue": "NCFCOPY"
                                },
                                {
                                    "code": "board",
                                    "label": "Board",
                                    "defaultValue": "NCERT"
                                },
                                {
                                    "code": "medium",
                                    "label": "Medium",
                                    "defaultValue": [
                                        "English"
                                    ]
                                }
                            ]
                        },
                        "groupBy": {
                            "value": "subject",
                            "defaultValue": "subject"
                        },
                        "collectionList": [],
                        "collectionType": "Textbook"
                    },
                    "description": "",
                    "publishedDate": ""
                },
                {
                    "id": "ng.sunbird.chapterList",
                    "ver": "1.0",
                    "data": {},
                    "author": "Kartheek",
                    "compId": "chapterListComponent",
                    "config": {
                        "contentTypes": {}
                    },
                    "description": "",
                    "publishedDate": ""
                },
                {
                    "id": "ng.sunbird.uploadComponent",
                    "ver": "1.0",
                    "data": {},
                    "author": "Kartheek",
                    "compId": "uploadContentComponent",
                    "config": {
                        "resourceTitleLength": "200"
                    },
                    "description": "",
                    "publishedDate": ""
                },
                {
                    "id": "ng.sunbird.practiceSetComponent",
                    "ver": "1.0",
                    "data": {},
                    "author": "Kartheek",
                    "compId": "practiceSetComponent",
                    "config": {
                        "tenantName": "",
                        "assetConfig": {
                            "image": {
                                "size": "50",
                                "accepted": "jpeg, png, jpg"
                            },
                            "video": {
                                "size": "50",
                                "accepted": "pdf, mp4, webm, youtube"
                            }
                        },
                        "solutionType": [
                            "Video",
                            "Text & image"
                        ],
                        "No of options": 4,
                        "questionCategory": [
                            "vsa",
                            "sa",
                            "ls",
                            "mcq",
                            "curiosity"
                        ],
                        "resourceTitleLength": "200"
                    },
                    "description": "",
                    "publishedDate": ""
                },
                {
                    "id": "ng.sunbird.dashboard",
                    "ver": "1.0",
                    "data": {},
                    "author": "Venkanna Gouda",
                    "compId": "dashboardComp",
                    "config": {},
                    "description": "",
                    "publishedDate": ""
                }
            ],
            "gradeLevel": [],
            "blueprintMap": {},
            "loginReqired": true,
            "sharedContext": [
                "channel",
                "framework",
                "board",
                "medium",
                "gradeLevel",
                "subject",
                "boardIds",
                "mediumIds",
                "gradeLevelIds",
                "subjectIds",
                "topic",
                "learningOutcome",
                "learningOutcomeIds"
            ],
            "defaultContributeOrgReview": true
        },
        "rolemapping": null,
        "createdby": "a61f23e7-b448-413b-aa36-3254dab503df",
        "updatedby": null,
        "createdon": "2023-03-27T04:54:23.161Z",
        "updatedon": "2023-03-27T04:54:22.673Z",
        "rootorg_id": "0137541424673095687",
        "sourcing_org_name": "Sunbird Org",
        "channel": "DIKSHA",
        "template_id": "template1",
        "guidelines_url": null,
        "acceptedcontents": null,
        "rejectedcontents": null,
        "sourcingrejectedcomments": null
    }
}'

Console screen shot
![Screenshot from 2023-04-04 16-05-18](https://user-images.githubusercontent.com/127090959/229766734-b9d603ed-f979-4bbd-a1e0-1cd1b5b2fc50.png)

**Reply by shubhambansaltarento** [reply]
This is my DB Connection 
![Screenshot from 2023-04-05 10-39-53](https://user-images.githubusercontent.com/127090959/229986535-b063f9f1-7bc1-4e4c-a110-7e707439bea4.png)

**Comment by vaivk369** [comment]
Here is the Installation Document link : https://cokreat.sunbird.org/use/developer-installation/contribution-service. Please check if you followed the steps and updated environment file accordingly.

**Reply by shubhambansaltarento** [reply]
@vaivk369 I have done this setup already and my program service is running but when I try to create a program using above curl request no response is coming, I am unsure weather our DB connection in program service is working or not as it is working in a separate node project, can you please check and confirm weather my configuration and curl request is correct ?

**Reply by arun1205** [reply]
@shubhambansaltarento I believe u r using node v 12.x.x as mentioned in the documentation. use 12.20.1 to be precise

**Comment by vaivk369** [comment, marked as answer]
To conclude, the program service has to run on node version 12.x.x and the all the steps in the document https://cokreat.sunbird.org/use/developer-installation/contribution-service should be followed sequentially.
